blocks/variables_dynamic.js

```diff
@@ -102,11 +102,14 @@ Blockly.Constants.VariablesDynamic.CUSTOM_CONTEXT_MENU_VARIABLE_GETTER_SETTER_MI
     }
     var opposite_type;
     var contextMenuMsg;
+    var varType;
     if (this.type == 'variables_get_dynamic') {
       opposite_type = 'variables_set_dynamic';
+      varType = this.outputConnection.check_ ? this.outputConnection.check_[0] : "";
```

Comment:
Rather than getting the first item in the checks list, get the variable from the field and use its type.

Reply:
OK ~ done ~